sdk/core/azure-core/test/ut/service_tracing_test.cpp

```diff
@@ -88,6 +88,28 @@ TEST(TracingContextFactory, SimpleServiceSpanTests)
     EXPECT_FALSE(contextAndSpan.Context.IsCancelled());
   }
 }
+
+TEST(TracingContextFactory, DeprecatedFactoryCtorForServiceWhichReleasedWithThisDependency)
+{
+  Azure::Core::_internal::ClientOptions clientOptions;
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4996)
+#else
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#endif
+
+  Azure::Core::Tracing::_internal::TracingContextFactory serviceTrace(
+      clientOptions, "my.service", "1.0b2");
+
+#ifdef _MSC_VER
+#pragma warning(pop)
+#else
+#pragma clang diagnostic pop
+
+#endif // _MSC_VER
```

Comment:
```suggestion
#ifdef _MSC_VER
#pragma warning(push)
#pragma warning(disable : 4996)
#elif defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) // !_MSC_VER
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
#elif defined(__clang__) // !_MSC_VER !__clang__
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
#endif // _MSC_VER

#ifdef _MSC_VER
#else
#endif

  Azure::Core::Tracing::_internal::TracingContextFactory serviceTrace(
      clientOptions, "my.service", "1.0b2");

#ifdef _MSC_VER
#pragma warning(pop)
#elif defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) // !_MSC_VER
#pragma GCC diagnostic pop
#elif defined(__clang__) // !_MSC_VER !__clang__
#pragma clang diagnostic pop
#endif // _MSC_VER
```

Reply:
Already there - FWIW, the GNUC version checks don't appear to be required (they're not in other places in the code where we deal with GCC).